plugins/out_stackdriver/stackdriver.c

```diff
@@ -524,6 +637,147 @@ static int stackdriver_format(const void *data, size_t bytes,
       msgpack_pack_str_body(&mp_pck,
                             ctx->instance_id, flb_sds_len(ctx->instance_id));
     }
+    else if (strcmp(ctx->resource, "k8s_container") == 0) {
+      /* k8s_container resource has fields project_id, location, cluster_name,
+                                          namespace_name, pod_name, container_name */
+
+        ret = process_local_resource_id(data, bytes, ctx, "k8s_container");
+        if (ret != 0) {
+                
+            flb_plg_error(ctx->ins, "Fail to process local_resource_id from log entry");
+            msgpack_sbuffer_destroy(&mp_sbuf);
+            flb_sds_destroy(ctx->namespace_name);
```

Comment:
I think it would throw errors if ctx->namespace_name is nil. Did you test it before?

I suggest to destroy these fields in process_local_resource_id.

Reply:
I think the implementation of "flb_sds_destroy" will check if the input is nil of not.
```
void flb_sds_destroy(flb_sds_t s)
{
    struct flb_sds *head;

    if (!s) {
        return;
    }

    head = FLB_SDS_HEADER(s);
    flb_free(head);
}
```